From b915412845933b4d2f98c3966ada4f12776a0cb6 Mon Sep 17 00:00:00 2001
From: Harish Kasiviswanathan <Harish.Kasiviswanathan@amd.com>
Date: Wed, 8 Feb 2017 10:01:23 -0500
Subject: [PATCH 1955/2094] kernel: Export mm_access

To implement Cross memomry attach for ROCm we need to check the
permissions if the current process can access remote process memory. The
best way to check permissions is to call mm_access. Unfortunately, this
function is not exported.

Temporarily export this function to test CMA.

I will revert this change and provide a local version that will have
more restrictive permission.

Change-Id: Icbda2f8f84bfba71e4be785c112f7a1c46cbea39
Signed-off-by: Harish Kasiviswanathan <Harish.Kasiviswanathan@amd.com>
---
 kernel/fork.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/kernel/fork.c b/kernel/fork.c
index f1751cb..fc76aff 100644
--- a/kernel/fork.c
+++ b/kernel/fork.c
@@ -1018,6 +1018,7 @@ struct mm_struct *mm_access(struct task_struct *task, unsigned int mode)
 
 	return mm;
 }
+EXPORT_SYMBOL_GPL(mm_access);
 
 static void complete_vfork_done(struct task_struct *tsk)
 {
-- 
2.7.4

